client/state/sites/domains/types.ts

```diff
@@ -0,0 +1,38 @@
+export interface SiteDomain {
```

Comment:
Oh! Thank you for noticing that. I just assume that most of our endpoint response data is not typed currently, but I probably should have looked more deeply. 

It looks like the `SiteDomain` I added here does overlap considerably with `ResponseDomain` in the other PR. I think it would be fine to use that instead. In any case, this PR resolves some TS errors and it's more accurate than it was; we can dedupe the types in a later PR. (cc @stephanethomas since you added this code originally.)

Reply:
I've created this Asana card to track that improvement: 1200182182542585-as-1201427040510440 👍.